src/domain/flow-system/domain/src/entities/flow_configuration/flow_configuration_state.rs

```diff
@@ -27,6 +27,48 @@ impl FlowConfigurationState {
     pub fn is_active(&self) -> bool {
         self.status.is_active()
     }
+
+    pub fn try_get_schedule(self) -> Option<Schedule> {
+        match self.rule {
+            FlowConfigurationRule::Schedule(schedule) => Some(schedule),
+            FlowConfigurationRule::IngestRule(ingest) => Some(ingest.schedule_condition),
+            FlowConfigurationRule::CompactionRule(_)
+            | FlowConfigurationRule::ResetRule(_)
+            | FlowConfigurationRule::TransformRule(_) => None,
+        }
+    }
+
+    pub fn try_get_ingest_rule(self) -> Option<IngestRule> {
+        if let FlowConfigurationRule::IngestRule(ingest_rule) = self.rule {
```

Comment:
I don't think it's a reasonable suggestion. There are no scenarios when ingest rule will be available from another enum choice. Besides, it would be really tricky to implement a new flow without updating this file :-)